aws/aws-parallelcluster-cookbook
# Move manage_raid resource to environment and use volume resource on it.

### Description of changes
 * Move `manage_raid` resource to environment cookbook
 * Use `volume` resource to manage raid
   * Move wait_for_block_dev to environment. It is used by both volume and raid resources.
  * Adapt ChefSpec accordingly
  * Mark `raid_vol_array` as mandatory for mount and unmount actions only

### Tests
* Move ebs_mount hooks to config folder
* Fix ebs device check. nvme1n1 was added there as a placeholder, it was wrong.
* Add Inspec tests for RAID resource
* Define hooks for mount and unmount raid tests
* Add lifecycle hooks for ebs_unmount test

Executed:
* `bash kitchen.ec2.sh environment-config test raid-mount-alinux2`
* `bash kitchen.ec2.sh environment-config test raid-unmount-alinux2`
* `bash kitchen.ec2.sh environment-config test raid-unexport-alinux2`
* `bash kitchen.ec2.sh environment-config test ebs-unmount-alinux2`
